plugins/plugin-chart-echarts/src/BoxPlot/transformProps.ts

```diff
@@ -0,0 +1,178 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import {
+  CategoricalColorNamespace,
+  ChartProps,
+  convertMetric,
+  DataRecord,
+  getNumberFormatter,
+  QueryFormDataMetric,
+} from '@superset-ui/core';
+import { BoxPlotQueryFormData } from './types';
+import { EchartsProps } from '../types';
+import { extractGroupbyLabel } from '../utils/series';
+
+export default function transformProps(chartProps: ChartProps): EchartsProps {
+  const { width, height, formData, queryData } = chartProps;
+  const data: DataRecord[] = queryData.data || [];
+  const {
+    colorScheme,
+    groupby = [],
+    metrics: formDataMetrics,
+    numberFormat,
+    xTicksLayout,
+  } = formData as BoxPlotQueryFormData;
+  const colorFn = CategoricalColorNamespace.getScale(colorScheme as string);
+  const numberFormatter = getNumberFormatter(numberFormat);
+
+  // TODO: remove cast once metrics is cleaned up in QueryFormData
+  const metrics = (formDataMetrics as QueryFormDataMetric[]).map(
+    metric => convertMetric(metric).label,
+  );
+
+  const transformedData = data
+    .map(datum => {
+      return metrics.map(metric => {
+        const groupbyLabel = extractGroupbyLabel({ datum, groupby });
+        const name = metrics.length === 1 ? groupbyLabel : `${groupbyLabel}, ${metric}`;
+        return {
+          name,
+          value: [
+            datum[`${metric}__low`],
+            datum[`${metric}__q1`],
+            datum[`${metric}__median`],
+            datum[`${metric}__q3`],
+            datum[`${metric}__high`],
+            datum[`${metric}__mean`],
+            datum[`${metric}__count`],
+            datum[`${metric}__outliers`],
+          ],
+          itemStyle: {
+            color: colorFn(groupbyLabel),
+            opacity: 0.6,
+            borderColor: colorFn(groupbyLabel),
+          },
+        };
+      });
+    })
+    .flatMap(row => row);
+
+  const outlierData = data
+    .map(datum => {
+      return metrics.map(metric => {
+        const groupbyLabel = extractGroupbyLabel({ datum, groupby });
+        const name = metrics.length === 1 ? groupbyLabel : `${groupbyLabel}, ${metric}`;
+        // Outlier data is a nested array of numbers (uncommon, therefore no need to add to DataRecordValue)
+        const outlierDatum = (datum[`${metric}__outliers`] || []) as number[];
+        return {
+          name: 'outlier',
+          type: 'scatter',
+          data: outlierDatum.map(val => [name, val]),
+          tooltip: {
+            formatter: (param: { data: [string, number] }) => {
+              const headline = `<p><strong>${param.data[0]}</strong></p>`;
```

Comment:
I'm not sure what the correct way of doing this is (I tried `<b>xxx</b>` but that went all crazy in Explore view).